lizard_connector/connector.py

```diff
@@ -0,0 +1,342 @@
+            When set to False only the first page is obtained on get.
+        """
+        self.all_pages = all_pages
+        self.max_results = max_results
+        self.next_url = None
+        self.count = None
+        self.username = username
+        self.password = password
+
+    def get(self, url):
+        """
+        GET a json from the api.
+        :param url: Lizard-api valid url.
+        :return: a dictionary of the 'results'-part of the api-response.
+        """
+        json_ = self.request(url)
+        self.count = json_.get('count')
+        self.next_url = json_.get('next')
+        json_ = json_.get('results', json_)
+        count = self.count if self.count else 0
+
+        if count > self.max_results:
+            raise LizardApiTooManyResults(
+                'Too many results: {} found, while max {} are accepted'.format(
+                json_.get('count', '?'), self.max_results)
+            )
+        if self.all_pages:
+            for extra_json in self:
+                json_.update(extra_json.get('results', extra_json))
+        return json_
+
+    def post(self, url, uuid, data):
+        """
+        POST data to the api.
+        :param url: Lizard-api valid endpoint url.
+        :param uuid: UUID of the object in the database you wish to store
+                     data to.
+        :param data: Dictionary with the data to post to the api
+        """
+        post_url = urllib.parse.urljoin(urllib.parse.urljoin(url, str(uuid)),
+                                        'data')
+        self.request(post_url, data=json.dumps(data))
+
+    def request(self, url, data=None):
+        """
+        GETs parameters from the Lizard api or POSTs data to the Lizard api.
+        Defaults to GET request. Turns into a POST request if data is provided.
+
+        :param url: full query url: should be of the form:
+                    [base_url]/api/v2/[endpoint]/?[query_key]=[query_value]&...
+        :param data: data in a list or dictionary format.
+        :return: the JSON from the response when no data is sent, else None.
+        """
+        if data:
+            request_obj = urllib.request.Request(url, headers=self.header,
+                                                 data=data)
+        else:
+            request_obj = urllib.request.Request(url, headers=self.header)
+        with urllib.request.urlopen(request_obj) as resp:
+            if not data:
+                encoding = resp.headers.get_content_charset()
+                encoding = encoding if encoding else 'UTF-8'
+                content = resp.read().decode(encoding)
+                return json.loads(content)
+
+    def next_page(self):
+        """
+        Returns next page if available else raises StopIteration.
+        """
+        return next(self)
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        try:
+            return self.request(self.next_url)
```

Comment:
Ik heb het idee dat dit een beetje van impliciete zaken aan elkaar hangt. ALS er "stiekum" een next_url is gezet op self DAN werkt het. En de `.get()` haalt niet daadwerkelijk alles op maar zet een next_url en dan moet je vervolgens weten dat je het als een iterator aan moet roepen.

Qua naamgeving is dat initieel niet duidelijk. En ik blijf die "return self" in `__iter__()` er raar uit blijven zien, hoewel dat misschien wel klopt. Maar dan zou ik wel verwachten dat er ergens duidelijk 'iterator" in de class naam staat, bijvoorbeeld.